runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1810,7 +1830,8 @@ TR::Node *TR_VectorAPIExpansion::fromBitsCoercedIntrinsicHandler(TR_VectorAPIExp
 
    TR::Node *newNode;
 
-   switch (elementType) {
+   int32_t type = mask ? TR::Int64 : elementType;
+   switch (type) {
```

Comment:
Will we still generate `vsplats`? Maybe we need to introduce `msplats`?

Reply:
When broadcastType == MODE_BITS_COERCED_LONG_TO_MASK, opcode we generate is `mLongBitsToMask`. In that case, the source type of the conversion must be long instead of the element type.

Reply:
Oh, right, of course, we do generate the right opcode! Then everything looks good to me.